### ISSUE TITLE
Remove WINDOW functions clauses from Rule L054 

### ISSUE THREAD

**Issue by cherns31** [sqlfluff/sqlfluff#2453]
SQLFluff is saying I’m violating L054, cause I’ve got implicit ORDER BY clauses, but EXPLICIT ones inside ORDER BY of my WINDOW Functions. I'd suggest that Window functions clauses should be excluded from this rule since, although feasible, it's unnatural to use IMPLICIT clauses within them.

```sql
SELECT 
    field_1
    , field_2
    , SUM(field_3) as field_3_total
    , SUM(field_3) OVER (ORDER BY field_1) AS field_3_window_sum
FROM table
GROUP BY 1, 2
ORDER BY 1, 2
```
